src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,261 @@
+      IconButton({ hiddenWhenEditing: false, icon: CloseIcon(), title: 'Finished Editing Reference' }),
+    ];
+  } else {
+    return [];
+  }
+};
+
+const IconButton = ({
+  hiddenWhenEditing,
+  title,
+  icon,
+}: {
+  hiddenWhenEditing: boolean;
+  title: string;
+  icon: ReactElement;
+}) => {
+  const { editing, setEditing } = useContext(EditingContext);
+  const hiddenClass = hiddenWhenEditing ? { hidden: editing } : { hidden: !editing };
+
+  return (
+    <button
+      className={cx(hiddenClass, 'float-right hover:text-cyan-100')}
+      key={title}
+      title={title}
+      onClick={() => {
+        setEditing(!editing);
+      }}
+    >
+      {icon}
+    </button>
+  );
+};
+
+export default function ReferenceDetailsCard({
+  reference,
+  handleReferenceChanged,
+}: {
+  reference: ReferenceItem;
+  handleReferenceChanged: (params: ReferenceItem) => void;
+}) {
+  const [editing, setEditing] = useState(false);
+  const cloneReferenceItem = <T extends object>(source: T): T => ({
+    ...source,
+  });
+
+  const updatedReference = cloneReferenceItem(reference);
+
+  // We have to use a copy of the reference for updating purposes, otherwise UpdateReferencesAtom sees no changes and doesn't update
+  const referenceChanged = useCallback(
+    (fieldName: string, value: string) => {
+      switch (fieldName) {
+        case 'citationKey':
+          updatedReference.citationKey = value;
+          break;
+        case 'title':
+          updatedReference.title = value;
+          break;
+        case 'doi':
+          updatedReference.doi = value;
+          break;
+      }
+      handleReferenceChanged(updatedReference);
+    },
+    [updatedReference, handleReferenceChanged],
+  );
+
+  const referenceDetailsCardFormat = {
+    citationKey: {
+      title: 'Citation Key',
+      editable: true,
+    },
+    title: {
+      title: 'Title',
+      editable: true,
+    },
+    authors: {
+      title: 'Authors',
+      editable: false,
+    },
+    doi: {
+      title: 'DOI',
+      editable: true,
+    },
+  };
+
+  return (
+    <EditingContext.Provider value={{ editing, setEditing }}>
+      <table
+        className={cx(
+          'm-2 h-max w-full ',
+          'border-collapse',
+          'bg-card-bg-primary',
+          'text-left',
+          'rounded',
+          'text-card-txt-primary',
+        )}
+      >
+        <thead>
+          <tr>
+            <TableHeadCell colSpan={2} content="References" header={true} />
+          </tr>
+        </thead>
+        <tbody className="divide-y">
+          {Object.entries(referenceDetailsCardFormat).map((row) => {
+            const key = row[0];
+            const rowData = row[1];
+            const contentData = reference[key as keyof ReferenceItem];
+
+            if (!contentData) {
+              return;
+            }
+
+            return (
+              <tr key={'row' + key + new Date().getTime().toString()}>
```

Comment:
Use `key={key}` directly or `key={'row' + key}` if you want the prefix (but we don't need it). You should not use the time because we want React to understand which row is being updated during re-renders. With the timestamp in the key, React would unmount and remount every row

Reply:
...
yeah turns out this was the thing causing the re-render to steal user focus after update. React is in fact smart enough not to re-render every row...now I know.